independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RemovedBeanImpl.java

```diff
@@ -14,16 +14,15 @@
     private final Set<Annotation> qualifiers;
 
     public RemovedBeanImpl(Kind kind, String description, Set<Type> types, Set<Annotation> qualifiers) {
-        this.kind = kind != null ? kind : Kind.CLASS;
+        this.kind = kind;
         this.description = description;
         this.types = CollectionHelpers.toImmutableSmallSet(types);
-        this.qualifiers = qualifiers != null ? CollectionHelpers.toImmutableSmallSet(qualifiers)
-                : Qualifiers.DEFAULT_QUALIFIERS;
+        this.qualifiers = CollectionHelpers.toImmutableSmallSet(qualifiers);
```

Comment:
I don't understand this, it looks lik you rather prefer to always allocate a new wrapping collection? I was trying to avoid those when not necessary.

Reply:
Note really, the `qualifiers` set is always null for default qualifiers and `CollectionHelpers.toImmutableSmallSet()` returns `null` for `null` param.